src/openforms/js/components/form/map.js

```diff
@@ -70,8 +72,26 @@ export default class Map extends TextFieldComponent {
 
     const map = L.map(`map-${this.id}`, MAP_DEFAULTS);
 
-    const {url: tileUrl, ...options} = TILE_LAYER_RD;
-    const tiles = L.tileLayer(tileUrl, options);
+    const tiles = L.tileLayer(
+      this.getTileLayerUrl(this.originalComponent?.tileLayerIdentifier),
+      TILE_LAYER_RD
+    );
     map.addLayer(tiles);
   }
+
+  // Try to get the tile layer url for the component.
+  // If it cannot be found, return the default url.
+  getTileLayerUrl(tileLayerIdentifier) {
+    const MAP_TILE_LAYERS = jsonScriptToVar('config-MAP_TILE_LAYERS', {
+      default: [],
+    });
```

Comment:
I'd move this outside of the method onto the module level - since the data is supposed to already be in the DOM before the JS starts to run, we can evaluate this at load-time rather than doing the same evaluation every time a map gets rendered.